src/display/api.js

```diff
@@ -910,7 +910,11 @@ class PDFDocumentProxy {
    * @returns {Promise} A promise that is resolved when clean-up has finished.
    */
   cleanup() {
-    return this._transport.startCleanup();
+    // Don't cleanup the font loader with Xfa since fonts are used in html
+    // to display Xfa content.
+    return this._transport.startCleanup(
+      /* cleanupFontLoader */ !this.isPureXfa
```

Comment:
With the other changes:
```suggestion
      /* keepFonts = */ this.isPureXfa
```